src/Text/Pandoc/Readers/Textile.hs

```diff
@@ -246,26 +258,47 @@ orderedListItemAtDepth = genericListItemAtDepth '#'
 -- | Common implementation of list items
 genericListItemAtDepth :: PandocMonad m => Char -> Int -> TextileParser m Blocks
 genericListItemAtDepth c depth = try $ do
-  count depth (char c) >> attributes >> whitespace
+  genericListItemStartAtDepth c depth
+  genericListItemContentsAtDepth depth
+
+-- | Lists always start with a number of leading characters '#' or '*'
+-- Ordered list start characters '#' can be followed by the start attribute
+-- number, but bullet list characters '*' can not
+genericListStartAtDepth :: PandocMonad m => Char -> Int -> TextileParser m Int
+genericListStartAtDepth c depth = do
+  count depth (char c)
+  case c of
+    '#' -> try orderedListStartAttr -- ordered list
+    _   -> return 1                 -- unordered list
+
+genericListItemStartAtDepth :: PandocMonad m => Char -> Int -> TextileParser m ()
+genericListItemStartAtDepth c depth =  () <$ (genericListStartAtDepth c depth >> attributes >> whitespace)
+
+genericListItemContentsAtDepth :: PandocMonad m => Int -> TextileParser m Blocks
+genericListItemContentsAtDepth depth = do
   contents <- mconcat <$> many ((B.plain . mconcat <$> many1 inline) <|>
                                 try (newline >> codeBlockHtml))
   newline
   sublist <- option mempty (anyListAtDepth (depth + 1))
   return $ contents <> sublist
 
+
 -- | A definition list is a set of consecutive definition items
 definitionList :: PandocMonad m => TextileParser m Blocks
 definitionList = try $ B.definitionList <$> many1 definitionListItem
 
 -- | List start character.
 listStart :: PandocMonad m => TextileParser m ()
 listStart = genericListStart '*'
-        <|> () <$ genericListStart '#'
+        <|> () <$ orderedListStart
         <|> () <$ definitionListStart
 
 genericListStart :: PandocMonad m => Char -> TextileParser m ()
 genericListStart c = () <$ try (many1 (char c) >> whitespace)
 
+orderedListStart :: PandocMonad m => TextileParser m ()
+orderedListStart = () <$ try (many1 (char '#') >> try orderedListStartAttr >> whitespace)
```

Comment:
Ah, so this confused me as well at first. 

`genericListStart` and `orderedListStart` are actually not used in `genericListItemAtDepth` (the one that parses and returns the list). Those two parsers are only used by `listStart`, which is then only used by `endline` here:

```haskell
-- | In Textile, an isolated endline character is a line break
endline :: PandocMonad m => TextileParser m Inlines
endline = try $ do
  newline
  notFollowedBy blankline
  notFollowedBy listStart
  notFollowedBy rawHtmlBlock
  return B.linebreak
```

I'm not familiar enough with that function to know whether or not we can safely remove the `try`. But I agree that there should be a shared Parser used by both `endline` and `genericListItemAtDepth`, since in theory they should match the same input.

However, it is a bit complicated to merge them directly, because of the `depth` argument. The "real" list parser needs to know the list level depth, but the `endline` parser does not.

Thoughts?

Reply:
Ah, you're right of course, I had missed that. I don't have a good solution either; please just ignore my comment.